src/java/boa/compiler/visitors/TypeCheckingVisitor.java

```diff
@@ -487,6 +489,17 @@ public void visit(final Factor n, final SymbolTable env) {
 						throw new TypeCheckException(n.getOperand(), e.getMessage(), e);
 					}
 					type = v.getFunction().erase(formalParameters);
+					if(((Identifier)n.getOperand()).getToken().equals("getValue")) {
```

Comment:
This feels like a nasty hack.  What is the purpose of these changes to the type checker?